### PR TITLE
SISRP-20822, add the Semesters card to the Student Overview page

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.berkeley.edu/browse/SISRP-20822

When viewing a student's semester card the advisor does not get links to class pages, etc.

### CODE REVIEW

**Comment by paulkerschen** [comment]
@johncrossman - A possible alternative to complicating the templates with conditionals would be to remove URL values from the My Academics feed as part of advisor filtering. That logic would need to loop through semesters, courses, and sections, scrubbing URLs from each.

File: src/assets/templates/academics_semesters.html [hunk 1]
```diff
@@ -6,10 +6,13 @@ <h2 class="cc-left">Semesters</h2>
   <div class="cc-widget-padding cc-flex-space-between-vertical-15">
     <div class="cc-table cc-academics-semester cc-academics-semester-{{semester.timeBucket}}" data-ng-repeat="semester in semesters | limitTo:pastSemestersLimit">
       <div>
-        <h3 data-ng-if="semester.hasEnrollmentData && semester.slug"><a class="cc-left" data-ng-href="/academics/semester/{{semester.slug}}" data-ng-bind="semester.name"></a></h3>
+        <!-- No link for advisors. -->
+        <h3 data-ng-if="semester.hasEnrollmentData && semester.slug && !isAdvisingStudentLookup"><a class="cc-left" data-ng-href="/academics/semester/{{semester.slug}}" data-ng-bind="semester.name"></a></h3>
+        <h3 data-ng-if="semester.hasEnrollmentData && semester.slug && isAdvisingStudentLookup" data-ng-bind="semester.name"></h3>
+
         <h3 data-ng-if="!semester.hasEnrollmentData || !semester.slug" data-ng-bind="semester.name"></h3>
```
Comment (paulkerschen) [inline src/assets/templates/academics_semesters.html]: @johncrossman Perhaps adding `|| isAdvisingStudentLookup` to this `data-ng-if` would be cleaner than introducing the new `h3` element on line 11?

Comment (paulkerschen) [inline src/assets/templates/academics_semesters.html]: @johncrossman Actually, as long as we're filtering the My Academics feed for use on the semesters card, deleting `slug` from each semester should have the same effect as a template-side conditional.

File: src/assets/javascripts/angular/controllers/pages/userOverviewController.js [hunk 2]
```diff
@@ -60,11 +60,9 @@ angular.module('calcentral.controllers').controller('UserOverviewController', fu
     advisingFactory.getStudentAcademics({
       uid: $routeParams.uid
     }).success(function(data) {
-      angular.extend($scope.academics, _.get(data, 'academics'));
-      $scope.collegeAndLevel = $scope.academics.collegeAndLevel;
-      $scope.examSchedule = $scope.academics.examSchedule;
```
Comment (johncrossman) [inline src/assets/javascripts/angular/controllers/pages/userOverviewController.js]: @paulkerschen Refactor on `/api/advising/academics/:student_uid`: the feed does not need top level 'academics' element. Removing it allows the removal of lines above.

File: public/dummy/json/advising_student_academics.json [hunk 3]
```diff
@@ -1,194 +1,165 @@
 {
-  "attributes":{
```
Comment (johncrossman) [inline public/dummy/json/advising_student_academics.json]: This dummy json was wrong in including non-academic data. It now resembles real world feed.

**Comment by paulkerschen** [comment]
Well filtered, @johncrossman, and thanks for extra dummy/fixture cleanup.

**Pull request merged**